core/src/main/scala/scalaz/std/List.scala

```diff
@@ -67,17 +67,11 @@ trait ListInstances extends ListInstances0 {
         }
         (a, b) => loop(a, b, Nil)
       }
-      def traverseImpl[F[_], A, B](l: List[A])(f: A => F[B])(implicit F: Applicative[F]) = {
```

Comment:
This crazy traverse blew the stack.

Reply:
In a single scientific test this builder based traverse ran about 15% faster than the original.

Reply:
Anecdotally I also find the replacement traverse to be faster than the old.